openedx_caliper_tracking/tasks.py

```diff
@@ -59,32 +63,37 @@ def _host_not_found(error, event, event_type):
 
     It would be called in case of "Host Not Found" error.
     """
-    LOGGER.error('Logs Delivery Failed: Could not deliver event ({}) to kafka because of {}.'.format(event_type, error))
-    sent_kafka_failure_email.delay(error, event, event_type)
+    LOGGER.error('Logs Delivery Failed: Could not deliver event ({}) to kafka ({}) because of {}.'.format(
+        event_type, _get_kafka_setting('END_POINT'), error.__class__.__name__))
+    sent_kafka_failure_email.delay(error.__class__.__name__)
 
 
 @task(bind=True)
-def sent_kafka_failure_email(self, error, event, event_type):
+def sent_kafka_failure_email(self, error):
     """
     Send error report to specified email address.
     """
-    additional_info = {
-        'Error': str(error),
-        'Event Type': event_type,
-        'Event': json.dumps(event, sort_keys=True)
-    }
+    if cache.get(CACHE_KEY):
+        LOGGER.info('Email Already Sent: Events delivrey failure report has been already sent to {}.'.format(
+            _get_kafka_setting('ERROR_REPORT_EMAIL')))
+        return
+
+    additional_info = {'Error': error}
     key = 'logs_not_sent'
     data = {
         'name': 'UCSD Support',
         'body': 'Below is the additional information regarding failure:',
         'additional_info': additional_info
     }
     subject = 'Failure in logs delivery to Kafka'
-    if send_notification(key, data, subject, DEFAULT_FROM_EMAIL, [ERROR_REPORT_EMAIL]):
-        success_message = 'Email Sent Succesfully: Event ({}) Delivery Failure Report Sent to {}.'.format(
-            event_type, ERROR_REPORT_EMAIL)
+    if send_notification(key, data, subject, DEFAULT_FROM_EMAIL, [_get_kafka_setting('ERROR_REPORT_EMAIL')]):
+        success_message = 'Email Sent Succesfully: Events delivery failure report sent to {}.'.format(
+            KafkaSettings.get_error_report_email())
```

Comment:
Can't we use `_get_kafka_setting` here as well ?

Reply:
Thanks for pointing it out. I just skipped it mistakenly.

Reply:
done in the latest commit.